tensorboard/plugins/image/images_plugin.py

```diff
@@ -61,14 +64,57 @@ def get_plugin_apps(self):
 
   def is_active(self):
     """The images plugin is active iff any run has at least one relevant tag."""
+    if self._db_connection_provider:
+      # The plugin is active if one relevant tag can be found in the database.
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          '''
+          SELECT 1
+          FROM Tags
+          WHERE Tags.plugin_name = ?
+          LIMIT 1
+          ''',
+          (metadata.PLUGIN_NAME,))
+      return bool(list(cursor))
     if not self._multiplexer:
       return False
     return bool(self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME))
 
   def _index_impl(self):
+    if self._db_connection_provider:
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          '''
+          SELECT
+            Runs.run_name,
+            Tags.tag_name,
+            Tags.display_name,
+            Descriptions.description,
+            /* Subtract 2 for leading width and height elements. */
+            MAX(CAST (Tensors.shape AS INT)) - 2 AS samples
+          FROM Tags
+          JOIN Runs USING (run_id)
+          JOIN Descriptions ON Tags.tag_id = Descriptions.id
+          JOIN Tensors ON Tags.tag_id = Tensors.series
+          WHERE Tags.plugin_name = :plugin
+            /* Shape should correspond to a rank-1 tensor. */
```

Comment:
I tend to avoid unclosed line comments when inside embedded string contexts - it's too easy for the string's newlines to get stripped in processing, and this can lead to extremely frustrating bugs if e.g. you have a line comment above an ORDER BY or the last 'AND <expr>' in a WHERE clause and then the newline gets stripped and you end up silently disabling part of the query.

I agree aesthetically `--` is a little nicer, but I think practically speaking it's worse here for that reason.